File: practica1/Ejercicio3/src/ejercicio3/Director.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ejercicio3;

/**
 *
 * @author carme
 */
public class Director {
    EmpleadoBuilder builder;
    Director(EmpleadoBuilder builder)
    {
        this.builder = builder;
    }
    
    public Empleado buildEmpleado(String nombre, String DNI, String cargo) {
        this.builder.build(nombre, DNI, cargo);
        return this.builder.getEmpleado();
    }
}

File: practica2/ejercicio_grupal/lib/director.dart
import 'empleadoBuilder.dart';
import 'empleado.dart';
import 'elementoEmpresa.dart';
import 'Departamento.dart';

class Director {
  late EmpleadoBuilder builder;
  late ElementoEmpresa empresa;

  Director(EmpleadoBuilder builder)
  {
    this.builder = builder;
    empresa = new Departamento('Empresa', null);
  }

  void addEmpleado(String nombre, String dni, String cargo, ElementoEmpresa? superior) {
    builder.build(nombre, dni, cargo, superior);
    if(superior == null)
      empresa.addElementoEmpresa(builder.getEmpleado());
  }

  void addDepartamento(String nombre, ElementoEmpresa? superior){
    empresa.addElementoEmpresa(Departamento(nombre, superior));
  }

  void remove(ElementoEmpresa elemento){
    empresa.removeElementoEmpresa(elemento);
  }

 ElementoEmpresa getEmpresa(){
    return empresa;
  }

}